tests/draft2019-09/additionalItems.json

```diff
@@ -93,5 +93,21 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "additionalItems should not look in applicators",
```

Comment:
I'm not sure this test is exactly the one that was intended to be added (maybe it is, don't know).

But right now this test essentially would pass because `additionalItems` without `items` in the same object does nothing.

That's a bit different from a test that tests `additionalItems` where it *does* do something, and where there's `items` present in the same object, but where there's other additional items present in an `allOf`'s schema.

Maybe we should add a test for that too, unless you see one already? Such a test should have as result `valid: false` (i.e. it should be possible to design one as such -- have a top level items that only constrains 1 item, have a allOf items constrain 2 items, then validate a 2-item array where the second item doesn't satisfy the additionalItems schema).

Let me know if that makes sense.

Reply:
If the test is instead:
```json
"schema": {
    "items": [ { "type": "integer" } ],
    "additionalItems": { "type": "boolean" }
}
"data": [ 1, null ]
```

The test would fail because "items" requires the first item to be an integer and, since there's only one item in there, that the remaining follow "additionalItems". The second item is "null", and so would fail "additionalItems".

Instead, this test wraps "Items" inside "allOf", so "additionalItems" has no visibility and that second "null" will pass.

The test, I believe, verifies that "additionalItems" can't peer inside "allOf". @gregsdennis thoughts?

Reply:
I'll add that having an "items" as a sibling to "additionalItems" defeats, in my opinion, the purpose of the test, where there's only an "additionalItems" inside an "allOf". Having only one guarantees that we're considering that "additionalItems" doesn't interact with that one.

Reply:
I just added a new test with a "valid=false" case.

Reply:
Think what you added covers what I meant, thanks!